RestaurantReservation.API/Authentication/JwtTokenGenerator.cs

```
@@ -0,0 +1,49 @@
+﻿using Microsoft.Extensions.Options;
+using RestaurantReservation.API.Models.Authentication;
+using System.IdentityModel.Tokens.Jwt;
+
+
+
+namespace RestaurantReservation.API.Authentication
+{
+    public class JwtTokenGenerator : IJwtTokenService
+    {
+        private readonly SigningConfiguration _signingConfiguration;
+        private readonly JwtConfiguration _jwtConfiguration;
+        private readonly IConfiguration _configuration;
+
+        public JwtTokenGenerator(SigningConfiguration signingConfiguration, IOptions<JwtConfiguration> jwtConfiguration, IConfiguration configuration)
+        {
+            _signingConfiguration = signingConfiguration;
+            _jwtConfiguration = jwtConfiguration?.Value ?? throw new ArgumentNullException(nameof(jwtConfiguration), "JwtConfiguration must not be null.");
+            _configuration = configuration;
+        }
+
+        public async Task<string> TryGenerateToken(AuthenticationCredentials authenticationCredentials)
+        {
+            var securityKey = _configuration["AuthenticationUserKey"];
+            var isValid = await Task.Run(() => ValidateCredentials(securityKey, authenticationCredentials.Key));
+
+            if (!isValid)
+            {
+                return null;
```

Let's throw an exception here or return an identified error somehow.